dp_check/dp_check.go

```diff
@@ -246,42 +259,73 @@ func main() {
 		}
 		req.Header.Add("Metadata-Flavor", "Google")
 		resp, err := client.Do(req)
-		infoLog.Printf("Check connection to metadata server by sending http GET request to %s", metadataServerUrl)
 		if err != nil {
 			return err
 		}
-		if resp.StatusCode != 200 {
-			return fmt.Errorf(`Connecting to metadata server failed. Either this VM doesn't have DirectPath access, or there is a bug that may be causing a larger outage`)
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
 		}
-		return nil
-	})
+		IPv6FromMetadataServer = net.ParseIP(strings.TrimSuffix(string(body), "\n"))
 
-	// Check IPv6
-	runCheck("IPv6 routes", func() error {
-		var out string
-		var err error
-		cmd := "ip -6 route show"
-		infoLog.Printf("Check IPv6 routes with:|%v|...\n", cmd)
-		if out, err = getOutput(cmd); err != nil {
+		infoLog.Printf("Check if this VM enables DirectPath by sending http GET request to %s", metadataServerUrl)
+		if err != nil {
 			return err
 		}
-		if !strings.Contains(out, "2001:4860:8040::/42") {
-			return fmt.Errorf(`Missing route prefix to backends: 2001:4860:8040::/42.
-IPv6 route setup either failed or hasn't been attempted`)
+		if resp.StatusCode == 200 {
+			return nil
 		}
-		return nil
+		if resp.StatusCode == 404 {
+			return fmt.Errorf("This VM doesn't have DirectPath access")
+		}
+		return fmt.Errorf("Connecting to metadata server failed. There may be a bug that may be causing a larger outage")
 	})
+
 	runCheck("IPv6 addresses", func() error {
-		var out string
-		var err error
-		cmd := "ip -6 addr show"
-		infoLog.Printf("Check IPv6 addresses with:|%v|...\n", cmd)
-		if out, err = getOutput(cmd); err != nil {
+		ifaces, err := net.Interfaces()
+		if err != nil {
 			return err
 		}
-		if !strings.Contains(out, "inet6 2600") {
+		// Go through all interfaces on the VM to see if IPv6 is enabled and if there is an IPv6 address, then check against the IPv6 address returned from metadataserver
+		var hasIPv6 bool
+		for _, iface := range ifaces {
```

Comment:
can we log details about the interfaces that were found? E.g. in a similar manner as to how `ip addr/route show` would display? Would be useful to know if someone e.g. was running this tool and sharing the log

Reply:
nit: though we're logging extra details in the case that we found a valid interface, can we also log verbose details for all interfaces that were found? Such log info would be useful for debugging if someone was running this tool and failing this check